test-harness/src/test/java/com/netflix/conductor/tests/integration/AbstractWorkflowServiceTest.java

```diff
@@ -5272,64 +5274,43 @@ public void testLambda() {
 
     @Test
     public void testTerminateTaskWithCompletedStatus() {
-        WorkflowDef workflowDef = new WorkflowDef();
-        workflowDef.setName("test_terminate_task_wf");
-        workflowDef.setSchemaVersion(2);
-
-        Map<String, Object> lambdaTaskInputParams = new HashMap<>();
-        lambdaTaskInputParams.put("input", "${workflow.input}");
-        lambdaTaskInputParams.put("scriptExpression", "if ($.input.a==1){return {testvalue: true}} else{return {testvalue: false}}");
+    	testTerminateTask(WorkflowStatus.COMPLETED);
+    }
 
-        WorkflowTask lambdaWorkflowTask = new WorkflowTask();
-        lambdaWorkflowTask.setWorkflowTaskType(TaskType.LAMBDA);
-        lambdaWorkflowTask.setName("lambda");
-        lambdaWorkflowTask.setInputParameters(lambdaTaskInputParams);
-        lambdaWorkflowTask.setTaskReferenceName("lambda0");
+    @Test
+    public void testTerminateTaskWithFailedStatus() {
+    	testTerminateTask(WorkflowStatus.FAILED);
+    }
 
-        Map<String, Object> terminateTaskInputParams = new HashMap<>();
-        terminateTaskInputParams.put(Terminate.getTerminationStatusParameter(), "COMPLETED");
-        terminateTaskInputParams.put(Terminate.getTerminationWorkflowOutputParameter(), "${lambda0.output}");
+    private WorkflowDef createSubworkflowDefWithWaitTask() {
```

Comment:
Since this is a workflow with wait task, this method can be named `createWorkflowWithWaitTask()` for better clarity.

Reply:
Also, please add another task after the wait task in this workflow, and check that this task is not scheduled when parent workflow is completed using terminate.

Reply:
made the requested changes